### PR TITLE
Add Local Filesystem Support

### PR DESCRIPTION
**Pull request by jacobbieker**
# Pull Request

## Description

Allows nowcasting-dataset to read files locally, without needing to get them from GCS or AWS.

Relates to issue #121 

## How Has This Been Tested?

Unit test

- [ ] No
- [x] Yes

## Checklist:

- [x] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/nowcasting/blob/main/coding_style.md)
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by JackKelly** [comment]
Nice!  And good timing, as we'll need this next week to get the code running on OCF ML server :)

File: nowcasting_dataset/dataset/datasets.py [hunk 1]
```diff
@@ -159,15 +159,15 @@ def __init__(
         self.gcs = None
         self.s3_resource = None
 
-        assert cloud in ["gcp", "aws"]
+        assert cloud in ["gcp", "aws", "local"]
```
Comment (JackKelly) [inline nowcasting_dataset/dataset/datasets.py:162]: I wonder if we should rename `cloud` to `compute_environment` or something like that?  `cloud = 'local'` feels a bit weird.  I don't have a strong opinion though!

**Review by JackKelly — APPROVED** [review]
Looks good to me!  (Although I admit I haven't yet fully wrapped my head around what's required to get `nowcasting_dataset` running locally!)

File: nowcasting_dataset/dataset/datasets.py (hunk 1 above)
Comment (jacobbieker) [reply nowcasting_dataset/dataset/datasets.py:162]: If we are going for being explicit, then probably should be `compute_environment`, but our own stuff is still just our own local `cloud` too!

Comment (JackKelly) [reply nowcasting_dataset/dataset/datasets.py:162]: Yeah, true!  So, yeah, I really don't mind what we do here.  Very happy to stick with `cloud`!

**Pull request merged**